### ISSUE TITLE
Consider changing BankingScheduledPayment to allow the capture of multiple nicknames and payeeReferences

### ISSUE THREAD

**Issue by anzbankau** [ConsumerDataStandardsAustralia/standards-maintenance#301]
ANZ would like to request a change to the standards for scheduled payments.

Currently the standards have a facility at the top level to capture nickname and payeeReference. The issue we are seeing is that when there is a multi-party payment, ANZ gives the customer the ability to set a nickname and payeeReference at the party level. Currently the standards don’t cater for this so we suggest adding optional fields to capture these details at the paymentSet level.

This is not an urgent change, just something to be considered for a future standards release.

**Comment by WestpacOpenBanking** [comment]
We believe that the change requested is conceptually correct and are supportive from that perspective. We note that the transition arrangements would need careful consideration and that participants would require sufficient time to implement – the change is breaking.

**Comment by CDR-API-Stream** [comment]
### Current Situation
Scheduled payments only allow for a global payee reference and payee nickname to be set regardless of the number of payment instructions in the payment set which each might represent different transactions to different payees. As such, banks cannot cater for different payee references and payee nicknames for each transaction.

### Problem Statement
Currently the payee reference and the nickname, acting as short display name for the payee, are required when describing a scheduled payment. However, many banks allow their customers to enter different payee references for each payment instruction in the scheduled payment's payment set. This would not preclude a common payee reference, but if present, it should be superseded by references for each payment in the payment set.

Further to this, the intention of nickname was a display name for the scheduled payment itself not the payee, so making this clear is important.

### Options

Based on the original change request, the following options have been identified.

**Option A - Make existing fields optional and introduce optional nickname and payeeReference at the paymentSet level**  
Make the existing nickname and payeeReference fields optional. This would allow banks to leave this data out if they don't support global instructions however it does not give ADR clients any guarantees on getting this data for the payment set.
Further, the nickname is incorrectly specified as the nickname of the payee whereas it is meant to be the nickname of the scheduled payment and making this optional would not be consistent with BankingPayee. Without the nickname for the scheduled payment being present, the ADR would not be able to infer the display name.

This is not ideal from a payload perspective as it puts more of the onus on ADRs to deal with missing data and allows for fields to be optional rather than mandatory and/or conditional however it represents an option that would not result in breaking changes and would result in less change management.

**Option B - Support individualised payee references**  
This change would allow for payee reference to be set as a global value but also overridden at the individual payment instruction within the payment set. Further to this, nickname would be corrected like Option A but set as mandatory to be consistent with BankingPayee and established norms on nicknames elsewhere int he standards.

This would allow the greatest flexibility and functional match whilst solving for current banking experiences. Both global and individual payment references would be supported. This would result in a breaking change and versioning of the end point.

#### Impacted Endpoints
* Get Scheduled Payments for Account v1
* Get Scheduled Payments Bulk v1
* Get Scheduled Payments For Specific Accounts v1

If Option B is preferred, the change would result in a v2 to all three end points.

### Option A analysis
This change deals with the solution as a non-breaking change. DHs would be free to modify their solutions within their own timeframes. The tradeoff is that the ADR may have to determine a suitable display name for scheduled payment nickname, payee nickname and payee reference because the changes do not require a nickname and payee reference - they remain optional. This would be a deviation from previous guidance on nicknames as a mechanism to represent a display name for the purposes of presenting a record - here a scheduled payment and payee - on the ADR side. Whilst they are optional they are considered SHOULD-level requirements. This may be supported through data conventions however wouldn't create a binding requirement. Changes to make are _italicised_:  

| Model | Field | Field Conditionality | Field Description | Change required |  
| :-- | :-- | :-- | :-- | :-- |  
| ``BankingScheduledPayment`` | ``payeeReference`` | _optional_ | The reference _, if applicable,_ for the transaction that will be used by the originating institution _for all payments in the payment set_ for the purposes of constructing a statement narrative on the payer’s account. Empty string if no data provided. | Change payeeReference to be optional and change too the description |  
| ``BankingScheduledPayment`` | ``nickname`` | optional | The short display name of the _scheduled payment_ as provided by the customer if provided. _Where a customer has not provided a nickname, a display name derived by the bank for the scheduled payment should be provided that is consistent with existing digital banking channels_ | Change nickname to be defined as the display name of the scheduled payment, not the payee and make it mandatory such that a default display name should apply in the absence of a customer supplied nickname. |
| ``BankingScheduledPaymentTo``	| _``payeeReference``_ | _optional_ | _The reference, if available, for the transaction that will be used by the originating institution for the given payment in the payment set for the purposes of constructing a statement narrative on the payer’s account. If not empty, it overrides the value provided at the ``BankingScheduledPayment`` level_ | Add ``payeeReference`` as a conditional field in ``BankingScheduledPaymentTo``. | The ``payeeReference`` at the ``BankingScheduledPayment`` level can be provided by the customer but if the payeeReference is provided within ``BankingScheduledPaymentTo`` this overrides the default set at the ``BankingScheduledPayment`` level.
| ``BankingScheduledPaymentTo``	| _``nickname``_ | _optional_ | _The short display name of the payee as provided by the customer unless ``toUType`` is set to ``payeeId``_. _Where a customer has not provided a nickname, a display name derived by the bank for payee should be provided that is consistent with existing digital banking channels_ | Add payee nickname as a mandatory field in BankingScheduledPaymentTo such that a default display name should apply in the absence of a customer supplied nickname. |

### Option B analysis
Because Option B presents a breaking change, further analysis is provided below.

To accommodate multiple payee references and nicknames, the following changes are proposed. Changes to to make are _italicised_:  

| Model | Field | Field Conditionality | Field Description | Change required |  
| :-- | :-- | :-- | :-- | :-- |  
| ``BankingScheduledPayment`` | ``payeeReference`` | _conditional_ | The reference _, if applicable,_ for the transaction that will be used by the originating institution _for all payments in the payment set_ for the purposes of constructing a statement narrative on the payer’s account. Empty string if no data provided.	| Change payeeReference to be conditional and description |  
| ``BankingScheduledPayment`` | ``nickname`` | _mandatory_ | The short display name of the _scheduled payment_ as provided by the customer if provided. _Where a customer has not provided a nickname, a display name derived by the bank for the scheduled payment consistent with existing digital banking channels_ | Change nickname to be defined as the display name of the scheduled payment, not the payee and make it mandatory such that a default display name would apply in the absence of a customer supplied nickname. |
| ``BankingScheduledPaymentTo``	| _``payeeReference``_ | _conditional_ | _The reference, if available, for the transaction that will be used by the originating institution for the given payment in the payment set for the purposes of constructing a statement narrative on the payer’s account. If not empty, it overrides the value provided at the ``BankingScheduledPayment`` level_ | Add ``payeeReference`` as a conditional field in ``BankingScheduledPaymentTo`` contingent on a payee reference being available for the specific payment. | The ``payeeReference`` at the ``BankingScheduledPayment`` level can be provided by the customer but if the payeeReference is provided within ``BankingScheduledPaymentTo`` this overrides the default set at the ``BankingScheduledPayment`` level.
| ``BankingScheduledPaymentTo``	| _``nickname``_ | _mandatory_	| _The short display name of the payee as provided by the customer unless ``toUType`` is set to ``payeeId``_. _Where a customer has not provided a nickname, a display name derived by the bank for payee consistent with existing digital banking channels_ | Add payee nickname as a mandatory field in BankingScheduledPaymentTo such that a default display name should apply in the absence of a customer supplied nickname. |

### Option B Implementation Considerations
This change is considered to be a breaking change because it proposes nickname to be mandatory and changes the conditionality for payeeReference.

**Key dates**
Scheduled payment APIs are required as part of Phase 2 obligations. Guidance on phasing is provided here: https://www.cdr.gov.au/rollout

The summary of key dates for Phase 2 is as follows:  

| Initial data holders (major banks) | Reciprocal DHs | Non-major ADIs and initial DH non-primary brands |  
| :-- | :-- | :-- |  
| 1st Nov 2020 | 1st July 2021 | 1st Nov 2021 |  

**Transition of obligations**

At a high level, there are three stages of transition:

| | Stage 1 &rarr; | Stage 2 &rarr; | Stage 3  
| :-- | :-- | :-- | :--  
| **Support** | Only v1 is required | Both v1 and v2 are both required | Only v2 is required
| **Data Holders** | Data Holders with Phase 2 obligations must support v1. They can support v2 at any point during this period | Data Holders with Phase 2 obligations must support both v1 and v2 of the affected end points. Affected DHs must be live with v2 from this date, but all DHs are encouraged to go earlier than this date to avoid compliance issues | From this stage, affected Data Holders can choose to retire v1 of the affected end points at their discretion
| **ADRs** | ADR clients can rely on v1 being available. v2 may be available for some DHs who go live early | ADR clients can rely on both v1 and v2 being available whilst they upgrade their client implementations to support v2 | From this stage onwards, ADRs cannot rely on v1 being available. Only v2 is guaranteed

**Options B future dated obligations**  

In considering the options, one of the primary aims is to allow the main wave of non-major ADIs to go live with only v2 from November 2021, thus avoiding the need to support both v1 and v2.

In this respect, two options have been identified for future dated obligations that consider the time required for ADRs to update their client implementations as well as the transition for data holders. These obligation dates align to Stage 2 and Stage 3 of the table above as well as the compliance obligations for phasing of data sharing against the data holder cohorts.

* **Option B1:** v2 obligations commence 1st March 2021 (commencement of stage 2). Retirement of v1 can occur from July 2021 (commencement of stage 3).  
* **Option B2:** v2 obligations commence 1st July 2021 (commencement of stage 2). Retirement of v1 can occur from Nov 2021 (commencement of stage 3).

The obligation dates are summarised in the following table:  

| | Stage 1 &rarr;<br/><br/>_(v1 required, v2 optional)_ | Stage 2 &rarr;<br/><br/>_(v1 required and v2 required)_ | Stage 3<br/><br/>_(v1 optional, v2 required)_ | Considerations |
| :-- | :-- | :-- | :-- | :-- |
| **Option B1** | Major banks support v1 and may support v2 | 1st March 2021<br/><br/_(v1 and v2 required from this date)_ | 1st July 2021<br/><br/>_(v1 may be retired from this date)_ | <ul><li>Initial data holders must support both v1 and v2 from March - July 2021.</li><li>Reciprocal DHs,  non-major ADIs and initial data holder non-primary brands only support v2</li><li>Non-major ADIs and initial data holder non-primary brands support only v2 from Nov 2021 (commencement of their Phase 2 obligations)</li><li>ADRs need to update their client implementations no later than 1st July 2021</ul> |
| **Option B2** | Major banks and Reciprocal DHs support v1 and may support v2 | 1st July 2021<br/><br/_(v1 and v2 required from this date)_ | 1st November 2021<br/><br/>_(v1 may be retired from this date)_ | <ul><li>Initial data holders continue to support v1 until November 2021</li><li>Initial data holders AND Reciprocal data holders must support both v1 and v2 from July - November 2021.</li><li>Non-major ADRs only have to support v2</li><li>Non-major ADIs and initial data holder non-primary brands support only v2 from Nov 2021 (commencement of their Phase 2 obligations)</li><li>ADRs need to update their client implementations no later than 1st November 2021</ul> |

### Summary

The DSB would welcome feedback on preferred option and feasible obligation dates if Option B is supported.

**Comment by anzbankau** [comment]
ANZ is supportive of Option 1.

However we do note that in the case of a multi-party scheduled payment we do not have an overall ‘nickname’ for the scheduled payment, so we would be looking to provide the individual payee nicknames at the BankingScheduledPaymentTo level.

From a timeline perspective we would preference having 6 months from the time the updated standards are available for data holders to adopt this.

**Comment by NationalAustraliaBank** [comment]
NAB’s preference is **Option A** – “Make existing fields optional and introduce optional nickname and payeeReference at the paymentSet level”
 
but we are not against **Option B** i.e., “Support individualised payee references” with the obligation dates mentioned in Option B2 – “v2 obligations commence 1st July 2021 (commencement of stage 2). Retirement of v1 can occur from Nov 2021 (commencement of stage 3)”.

**Comment by alex-histon** [comment]
Newcastle Permanent Building Society (non-major) supports Option A. We would also like to request that **payerReference** also be considered to add as a field at the paymentSet level, to support individualised payee references, as we support payers nominating different references on their own end as well.

**Comment by CDR-API-Stream** [comment]
The outcome of this issue has been [approved by the Data Standards Chair for inclusion in v1.7.0](https://github.com/ConsumerDataStandardsAustralia/standards/issues/159). A change to the data standards was recommended.

This issue will be closed accordingly.
